src/main/java/dev/revere/alley/base/server/ServerService.java

```diff
@@ -115,4 +148,73 @@ public void clearAllQueues(Player issuer) {
             }
         }
     }
+
+    @Override
+    public List<Material> getBlockedCraftingItems() {
+        return Collections.unmodifiableList(this.blockedCraftingItems);
+    }
+
+    @Override
+    public void loadBlockedCraftingMaterials() {
+        FileConfiguration config = this.configService.getSettingsConfig();
+        List<String> blocked = config.getStringList(this.BLOCKED_MATERIALS_SECTION);
+
+        this.blockedCraftingItems.clear();
+        for (String mat : blocked) {
+            try {
+                Material material = Material.valueOf(mat);
+                this.blockedCraftingItems.add(material);
+            } catch (IllegalArgumentException ignored) {
+                Bukkit.getLogger().warning("[ServerService] Invalid material in blocked list: " + mat);
+            }
+        }
+    }
+
+    @Override
+    public void saveCraftingRecipes(Material material) {
+        FileConfiguration config = this.configService.getSettingsConfig();
+        File configFile = this.configService.getConfigFile("settings.yml");
+
+        config.set(this.BLOCKED_MATERIALS_SECTION, this.blockedCraftingItems.stream().map(Material::name).collect(Collectors.toList()));
```

Comment:
readability improvement suggestion:
   
   List<String> materialNames = this.blockedCraftingItems.stream()
            .map(Material::name)
            .collect(Collectors.toList());

    config.set(this.BLOCKED_MATERIALS_SECTION, materialNames);